planner/core/logical_plan_builder.go

```diff
@@ -824,6 +824,21 @@ func (b *PlanBuilder) buildSort(p LogicalPlan, byItems []*ast.ByItem, aggMapper
 	sort := LogicalSort{}.Init(b.ctx)
 	exprs := make([]*ByItems, 0, len(byItems))
 	for _, item := range byItems {
+		switch x := item.Expr.(type) {
```

Comment:
Hmm, I prefer not adding this special conversion at this place, this looks pretty ad-hoc.

IMHO, this conversion should be encapsulated as part of `expressionRewriter` logically. We can pass in a flag like `inByClause` to `expressionRewriter` here, or in `resolveGbyExprs`, then this flag would work as context information of expression rewriting just like other fields in `expressionRewriter`, so when we are traversing the AST tree and meeting a `ParamMarkerExpr` in `expressionRewriter`, we check this flag to decide whether we should return a normal `Constant` or apply the above depicted logic of this patch.

Also, by encapsulating this logic into `expressionRewriter`, we can avoid redundant code for different kinds of `BY ?` clauses, including the `PARTITION BY ?` of upcoming window function.